File: src/main/java/org/galaxio/gatling/javaapi/utils/RandomPhoneGenerator.java
package org.galaxio.gatling.javaapi.utils;

import org.galaxio.gatling.utils.phone.*;

import java.util.List;

import static scala.jdk.javaapi.CollectionConverters.asScala;

public final class RandomPhoneGenerator {
    private RandomPhoneGenerator(){}

    public static String randomPhone(List<PhoneFormat> formats, TypePhone.TypePhone typePhone){
        return org.galaxio.gatling.utils.RandomPhoneGenerator.randomPhone(asScala(formats).toSeq(), typePhone);
    }

    public static String randomPhone(String pathToFormats, TypePhone.TypePhone typePhone){
        return org.galaxio.gatling.utils.RandomPhoneGenerator.randomPhone(pathToFormats, typePhone);
    }
}

File: src/main/scala/org/galaxio/gatling/utils/RandomPhoneGenerator.scala
package org.galaxio.gatling.utils

import org.galaxio.gatling.utils.phone.TypePhone._
import org.galaxio.gatling.utils.phone.{Phone, PhoneFormat}

object RandomPhoneGenerator {

  def randomPhone(
      formats: Seq[PhoneFormat],
      typePhone: TypePhone,
  ): String = typePhone match {
    case PhoneNumber         => Phone(formats).phoneNumber
    case TollFreePhoneNumber => Phone(formats).tollFreePhoneNumber
    case E164PhoneNumber     => Phone(formats).e164PhoneNumber
  }

  def randomPhone(
      pathToFormats: String,
      typePhone: TypePhone,
  ): String = typePhone match {
    case PhoneNumber         => Phone(pathToFormats).phoneNumber
    case TollFreePhoneNumber => Phone(pathToFormats).tollFreePhoneNumber
    case E164PhoneNumber     => Phone(pathToFormats).e164PhoneNumber
  }

}
